### PR TITLE
feat: wait 2 seconds before 409 retry

### PR DESCRIPTION
**Pull request by franciscoperezsammartino**
### What issues does this PR fix or reference?

@ W-12762068@

### Functionality Before

No wait between calls after 409

### Functionality After

Wait 2 seconds.

### CODE REVIEW

File: test/common/deployPipelineCache.test.ts [hunk 1]
```diff
@@ -63,10 +64,6 @@ describe('deployPipelineCache', () => {
     expect(excThrown);
   });
```
Comment (franciscoperezsammartino) [inline test/common/deployPipelineCache.test.ts:66]: This test was creating a similar function so just reused the same code from utils

File: test/commands/project/deploy/pipeline/start.test.ts [hunk 2]
```diff
@@ -639,7 +639,10 @@ describe('project deploy pipeline start', () => {
     describe('retry promotion request on 409 error', () => {
       const mockError = new Error();
 
+      let sleepStub: sinon.SinonStub;
+
       beforeEach(() => {
+        sleepStub = sinon.stub(Utils, 'sleep').resolves();
```
Comment (franciscoperezsammartino) [inline test/commands/project/deploy/pipeline/start.test.ts:645]: Im stubbing the sleep method so the tests dont have to wait for this.

File: test/commands/project/deploy/pipeline/start.test.ts [hunk 3]
```diff
@@ -738,6 +749,8 @@ describe('project deploy pipeline start', () => {
           expect(requestMock.callCount).to.equal(5);
           // make sure that we show the returned aor id to the user
           expect(ctx.stdout).to.contain('mock-aor-id');
+          sinon.assert.callCount(sleepStub, 4);
+          sinon.assert.alwaysCalledWith(sleepStub, 2000);
```
Comment (EstebanRomero84) [inline test/commands/project/deploy/pipeline/start.test.ts:753]: Very cool assert function.

**Review by EstebanRomero84 — APPROVED** [review]
This look good, thanks Fran.

**Pull request merged**